src/dataflow-types/src/types.rs

```diff
@@ -2173,6 +2173,31 @@ pub mod sources {
             }
         }
 
+        /// Returns `true` if this connector yields data that is
+        /// append-only/monotonic. Append-monly means the source
+        /// never produces retractions.
+        // TODO(guswynn): consider enforcing this more completely at the
+        // parsing/typechecking level, by not using an `envelope`
+        // for sources like pg
+        pub fn append_only(&self) -> bool {
```

Comment:
I know it's more typing but what do you think of being more explicit with our types in the match here? In particular I think it would provide value to force anyone who adds a `SourceEnvelope` / `SourceConnector` / `ExternalSourceConnector` variant to think about whether it is in fact append only

Reply:
great point!

Reply:
`or_patterns` stabilizing makes this much cleaner